### DISCUSSION TITLE
Possible to get original python type name? Ie distinguish between str and pathlib.Path

### DISCUSSION THREAD

**Discussion by Falmarri** [PyO3/pyo3#3720] [Questions] [answered]
Is there a way to get the original python type? For example if I have an api that takes an argument that could be either a string OR a path, I can't seem to find any way of distinguishing that information from rust.

**Comment by adamreichold** [comment, marked as answer]
Take `&PyAny` as your function argument and use `get_type` to get the associated type which as a `name` method.
